synapse/tools/backup.py

```diff
@@ -88,6 +88,9 @@ def txnbackup(lmdbinfo, srcdir, dstdir, skipdirs=None):
     srcdir = s_common.reqdir(srcdir)
     dstdir = s_common.gendir(dstdir)
 
+    # Copy cell.guid first to ensure the backup can be deleted
+    shutil.copy(os.path.join(srcdir, 'cell.guid'), os.path.join(dstdir, 'cell.guid'))
```

Comment:
Not sure this fixes ones that are already broken or lose the race between dir construction and this line?

Reply:
It doesn't, but we allow backing up to subdirs and checking for a `cell.guid` file is how `delBackup` and `getBackups` determine what is a backup or not. We could allow `delBackup` to just delete any folder that exists in the backup dir even if `getBackups` doesn't list it as a backup, but we do have tests for preventing deleting higher level dirs that would break.